org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingRecordPatternTest.java

```diff
@@ -585,4 +590,90 @@ public void testModifyRecordInstanceOfPattern() throws Exception {
 		assertEqualString(preview, buf.toString());
 	}
 
+	public void testNPEinASTRewriteFlattener() throws Exception {
+		if (checkAPILevel()) {
+			return;
+		}
+		IPackageFragment pack1= this.sourceFolder.createPackageFragment("test1", false, null);
+		StringBuilder buf= new StringBuilder();
+		buf.append("public class X {\n");
+		buf.append(		"void foo(Object o) {\n");
+		buf.append(		"	switch (o) {\n");
+		buf.append(	    "		default       	: System.out.println(\"0\");\n");
+		buf.append(	    "	}\n");
+		buf.append(	    "}\n");
+		buf.append(		"\n");
+		buf.append(		"}\n");
+
+		ICompilationUnit cu= pack1.createCompilationUnit("X.java", buf.toString(), false, null);
+
+		CompilationUnit astRoot= createAST(this.apiLevel, cu);
+		ASTRewrite rewrite= ASTRewrite.create(astRoot.getAST());
+
+		AST ast= astRoot.getAST();
+
+		assertTrue("Parse errors", (astRoot.getFlags() & ASTNode.MALFORMED) == 0);
+		TypeDeclaration type= findTypeDeclaration(astRoot, "X");
+		MethodDeclaration methodDecl= findMethodDeclaration(type, "foo");
+		Block block= methodDecl.getBody();
+		List blockStatements= block.statements();
+		assertTrue("Number of statements not 1", blockStatements.size() == 1);
+
+		{ // insert Guarded pattern
+			SwitchStatement switchStatement = (SwitchStatement) blockStatements.get(0);
+
+			List statements= switchStatement.statements();
+			assertTrue("Number of statements not 2", statements.size() == 2);
+
+			SwitchCase caseStatement= ast.newSwitchCase();
+			caseStatement.setSwitchLabeledRule(false);
+			GuardedPattern guardedPattern = ast.newGuardedPattern();
+			TypePattern typePattern = ast.newTypePattern();
+			SingleVariableDeclaration patternVariable = ast.newSingleVariableDeclaration();
+			patternVariable.setType(ast.newSimpleType(ast.newSimpleName("Integer")));
```

Comment:
Suby, as discussed these need to be rewritten - no overwriting required. Either this is sufficient, or have a source code with GP and then rewrite with null values

Reply:
I have modified the unit test